File: api/internal/model/team.go
package model

import "time"

// TeamStatus 团队状态
type TeamStatus string

const (
	TeamStatusActive   TeamStatus = "active"   // 正常
	TeamStatusBusy     TeamStatus = "busy"     // 忙碌（接单中）
	TeamStatusInactive TeamStatus = "inactive" // 停用
)

// TeamMemberRole 团队成员角色
type TeamMemberRole string

const (
	TeamMemberRoleLeader TeamMemberRole = "leader" // 队长
	TeamMemberRoleMember TeamMemberRole = "member" // 队员
)

// TeamMemberStatus 团队成员状态
type TeamMemberStatus string

const (
	TeamMemberStatusActive TeamMemberStatus = "active" // 正常
	TeamMemberStatusLeft   TeamMemberStatus = "left"   // 已离队
	TeamMemberStatusKicked TeamMemberStatus = "kicked" // 被踢出
)

// Team 陪玩师团队
type Team struct {
	Base
	Name        string     `json:"name" gorm:"size:64;not null"`                    // 团队名称
	Description string     `json:"description" gorm:"size:255"`                     // 团队简介
	AvatarURL   string     `json:"avatarUrl" gorm:"column:avatar_url;size:255"`     // 团队头像
	LeaderID    uint64     `json:"leaderId" gorm:"column:leader_id;not null;index"` // 队长ID（Player.ID）
	Status      TeamStatus `json:"status" gorm:"size:32;default:'active';index"`    // 状态

	// 配置
	MaxMembers  int `json:"maxMembers" gorm:"column:max_members;default:5"`   // 最大成员数
	MemberCount int `json:"memberCount" gorm:"column:member_count;default:1"` // 当前成员数

	// 收入分配配置
	IncomeShareType string  `json:"incomeShareType" gorm:"column:income_share_type;size:32;default:'equal'"` // equal=平分, custom=自定义
	LeaderBonusRate float64 `json:"leaderBonusRate" gorm:"column:leader_bonus_rate;default:0"`               // 队长额外抽成比例（预留）

	// 统计
	TotalOrderCount  int   `json:"totalOrderCount" gorm:"column:total_order_count;default:0"`   // 累计接单数
	TotalIncomeCents int64 `json:"totalIncomeCents" gorm:"column:total_income_cents;default:0"` // 累计收入（分）

	// 当前订单（忙碌状态时）
	CurrentOrderID *uint64 `json:"currentOrderId,omitempty" gorm:"column:current_order_id;index"` // 当前进行中的订单ID

	// Relations
	Leader  *Player      `json:"leader,omitempty" gorm:"foreignKey:LeaderID"`
	Members []TeamMember `json:"members,omitempty" gorm:"foreignKey:TeamID"`
}

// TableName 指定表名
func (Team) TableName() string {
	return "teams"
}

// IsBusy 是否忙碌
func (t *Team) IsBusy() bool {
	return t.Status == TeamStatusBusy || t.CurrentOrderID != nil
}

// CanAcceptOrder 是否可以接单
func (t *Team) CanAcceptOrder() bool {
	return t.Status == TeamStatusActive && t.CurrentOrderID == nil
}

// TeamMember 团队成员
type TeamMember struct {
	Base
	TeamID   uint64           `json:"teamId" gorm:"column:team_id;not null;index;uniqueIndex:idx_team_player"`     // 团队ID
	PlayerID uint64           `json:"playerId" gorm:"column:player_id;not null;index;uniqueIndex:idx_team_player"` // 陪玩师ID
	Role     TeamMemberRole   `json:"role" gorm:"size:32;default:'member'"`                                        // 角色
	Status   TeamMemberStatus `json:"status" gorm:"size:32;default:'active';index"`                                // 状态
	JoinedAt time.Time        `json:"joinedAt" gorm:"column:joined_at;not null"`                                   // 加入时间
	LeftAt   *time.Time       `json:"leftAt,omitempty" gorm:"column:left_at"`                                      // 离队时间

	// 顺位（用于队长继承）
	SortOrder int `json:"sortOrder" gorm:"column:sort_order;default:0"` // 排序（越小越靠前，队长继承顺序）

	// 统计
	OrderCount  int   `json:"orderCount" gorm:"column:order_count;default:0"`   // 在团队内接单数
	IncomeCents int64 `json:"incomeCents" gorm:"column:income_cents;default:0"` // 在团队内收入（分）

	// Relations
	Team   *Team   `json:"team,omitempty" gorm:"foreignKey:TeamID"`
	Player *Player `json:"player,omitempty" gorm:"foreignKey:PlayerID"`
}

// TableName 指定表名
func (TeamMember) TableName() string {
	return "team_members"
}

// IsLeader 是否是队长
func (tm *TeamMember) IsLeader() bool {
	return tm.Role == TeamMemberRoleLeader
}

// IsActive 是否在队
func (tm *TeamMember) IsActive() bool {
	return tm.Status == TeamMemberStatusActive
}

// TeamInviteStatus 团队邀请状态
type TeamInviteStatus string

const (
	TeamInviteStatusPending  TeamInviteStatus = "pending"  // 待处理
	TeamInviteStatusAccepted TeamInviteStatus = "accepted" // 已接受
	TeamInviteStatusRejected TeamInviteStatus = "rejected" // 已拒绝
	TeamInviteStatusExpired  TeamInviteStatus = "expired"  // 已过期
)

// TeamInvite 团队邀请（预留）
type TeamInvite struct {
	Base
	TeamID    uint64           `json:"teamId" gorm:"column:team_id;not null;index"`       // 团队ID
	PlayerID  uint64           `json:"playerId" gorm:"column:player_id;not null;index"`   // 被邀请的陪玩师ID
	InviterID uint64           `json:"inviterId" gorm:"column:inviter_id;not null;index"` // 邀请人ID
	Status    TeamInviteStatus `json:"status" gorm:"size:32;default:'pending';index"`     // 状态
	ExpireAt  time.Time        `json:"expireAt" gorm:"column:expire_at;not null;index"`   // 过期时间
	Message   string           `json:"message" gorm:"size:255"`                           // 邀请留言

	// Relations
	Team    *Team   `json:"team,omitempty" gorm:"foreignKey:TeamID"`
	Player  *Player `json:"player,omitempty" gorm:"foreignKey:PlayerID"`
	Inviter *Player `json:"inviter,omitempty" gorm:"foreignKey:InviterID"`
}

// TableName 指定表名
func (TeamInvite) TableName() string {
	return "team_invites"
}

File: admin/src/api/team.ts
import apiClient from './client';
import type { ApiResponse, PaginatedResponse } from '@/types/api';

// ============================================================================
// Types
// ============================================================================

export type TeamStatus = 'active' | 'busy' | 'inactive';

export type TeamMemberRole = 'leader' | 'member';

export type TeamMemberStatus = 'active' | 'left' | 'kicked';

export type TeamInviteStatus = 'pending' | 'accepted' | 'rejected' | 'expired';

export interface Team {
    id: number;
    name: string;
    description?: string;
    avatarUrl?: string;
    leaderId: number;
    leader?: {
        id: number;
        nickname: string;
        avatar?: string;
        rank?: string;
    };
    status: TeamStatus;
    maxMembers: number;
    memberCount: number;
    incomeShareType: 'equal' | 'custom';
    leaderBonusRate: number;
    totalOrderCount: number;
    totalIncomeCents: number;
    currentOrderId?: number;
    createdAt: string;
    updatedAt: string;
}

export interface TeamMember {
    id: number;
    teamId: number;
    playerId: number;
    player?: {
        id: number;
        nickname: string;
        avatar?: string;
        rank?: string;
    };
    role: TeamMemberRole;
    status: TeamMemberStatus;
    sortOrder: number;
    orderCount: number;
    incomeCents: number;
    joinedAt: string;
    leftAt?: string;
}

export interface TeamInvite {
    id: number;
    teamId: number;
    playerId: number;
    player?: {
        id: number;
        nickname: string;
        avatar?: string;
    };
    inviterId: number;
    inviter?: {
        id: number;
        nickname: string;
        avatar?: string;
    };
    status: TeamInviteStatus;
    expireAt: string;
    message?: string;
    createdAt: string;
    updatedAt: string;
}

export interface TeamStats {
    totalTeams: number;
    activeTeams: number;
    busyTeams: number;
    inactiveTeams: number;
    totalMembers: number;
    totalIncomeCents: number;
}

// ============================================================================
// Request/Response Types
// ============================================================================

export interface TeamListParams {
    page?: number;
    page_size?: number;
    keyword?: string;
    status?: TeamStatus;
    leaderId?: number;
    minMember?: number;
    maxMember?: number;
}

export interface TeamListResponse extends PaginatedResponse {
    items: Team[];
}

export interface TeamCreateRequest {
    name: string;
    description?: string;
    avatarUrl?: string;
    leaderPlayerId: number;
    maxMembers?: number;
    incomeShareType?: 'equal' | 'custom';
    leaderBonusRate?: number;
}

export interface TeamUpdateRequest {
    name: string;
    description?: string;
    avatarUrl?: string;
    maxMembers?: number;
    incomeShareType?: 'equal' | 'custom';
    leaderBonusRate?: number;
}

export interface TeamStatusUpdateRequest {
    status: TeamStatus;
}

export interface MemberListParams {
    page?: number;
    page_size?: number;
    teamId?: number;
    playerId?: number;
    role?: TeamMemberRole;
    status?: TeamMemberStatus;
}

export interface MemberListResponse extends PaginatedResponse {
    items: TeamMember[];
}

export interface AddMemberRequest {
    playerId: number;
}

export interface TransferLeaderRequest {
    newLeaderPlayerId: number;
}

export interface InviteListParams {
    page?: number;
    page_size?: number;
    teamId?: number;
    playerId?: number;
    status?: TeamInviteStatus;
}

export interface InviteListResponse extends PaginatedResponse {
    items: TeamInvite[];
}

// Batch operations

export interface BatchDeleteTeamsRequest {
    team_ids: number[];
}

export interface BatchUpdateTeamsStatusRequest {
    team_ids: number[];
    status: TeamStatus;
}

export interface BatchAddTeamMembersRequest {
    team_id: number;
    player_ids: number[];
}

export interface BatchOperationError {
    id: number;
    message: string;
}

export interface BatchOperationResponse {
    successCount: number;
    failedCount: number;
    totalCount: number;
    successItems: number[];
    failedItems: BatchOperationError[];
}

// ============================================================================
// API
// ============================================================================

export const teamApi = {
    // Team Management
    getTeams: (params?: TeamListParams) =>
        apiClient.get<ApiResponse<TeamListResponse>>('/admin/teams', { params }),

    getTeamDetail: (id: number) =>
        apiClient.get<ApiResponse<Team>>(`/admin/teams/${id}`),

    createTeam: (data: TeamCreateRequest) =>
        apiClient.post<ApiResponse<Team>>('/admin/teams', data),

    updateTeam: (id: number, data: TeamUpdateRequest) =>
        apiClient.put<ApiResponse<Team>>(`/admin/teams/${id}`, data),

    deleteTeam: (id: number) =>
        apiClient.delete<ApiResponse<void>>(`/admin/teams/${id}`),

    updateTeamStatus: (id: number, data: TeamStatusUpdateRequest) =>
        apiClient.put<ApiResponse<void>>(`/admin/teams/${id}/status`, data),

    getTeamStats: () =>
        apiClient.get<ApiResponse<TeamStats>>('/admin/teams/stats'),

    // Batch Operations
    batchDeleteTeams: (data: BatchDeleteTeamsRequest) =>
        apiClient.delete<ApiResponse<BatchOperationResponse>>('/admin/teams/batch', { data }),

    batchUpdateTeamsStatus: (data: BatchUpdateTeamsStatusRequest) =>
        apiClient.put<ApiResponse<BatchOperationResponse>>('/admin/teams/batch/status', data),

    batchAddTeamMembers: (data: BatchAddTeamMembersRequest) =>
        apiClient.post<ApiResponse<BatchOperationResponse>>('/admin/teams/batch/members', data),

    // Member Management
    getTeamMembers: (teamId: number) =>
        apiClient.get<ApiResponse<TeamMember[]>>(`/admin/teams/${teamId}/members`),

    addTeamMember: (teamId: number, data: AddMemberRequest) =>
        apiClient.post<ApiResponse<void>>(`/admin/teams/${teamId}/members`, data),

    removeTeamMember: (teamId: number, playerId: number) =>
        apiClient.delete<ApiResponse<void>>(`/admin/teams/${teamId}/members/${playerId}`),

    transferCaptain: (teamId: number, data: TransferLeaderRequest) =>
        apiClient.post<ApiResponse<void>>(`/admin/teams/${teamId}/transfer-leader`, data),

    // Global Member List
    listMembers: (params?: MemberListParams) =>
        apiClient.get<ApiResponse<MemberListResponse>>('/admin/teams/members', { params }),

    // Invite Management
    listInvites: (params?: InviteListParams) =>
        apiClient.get<ApiResponse<InviteListResponse>>('/admin/teams/invites', { params }),

    getInviteDetail: (inviteId: number) =>
        apiClient.get<ApiResponse<TeamInvite>>(`/admin/teams/invites/${inviteId}`),
};

export default teamApi;
